### ISSUE TITLE
Move post/page title to the top bar.

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#27093]
Block themes and template editing give authors much more freedom to express how the titles of posts and pages are displayed on the frontend. They might appear inside Cover blocks, adjacent to Post Content, or in any other conceivable block configuration. There are also situations in which a template might not display a post title at all, e.g. Post Formats.

All that to say, it might be time to consider _not_ displaying the post title within the canvas when editing a post or page, because the current implementation is increasingly unlikely to be an accurate representation of the frontend. Example:

| Backend | Frontend |
| --- | --- |
| <img width="1146" alt="Screenshot 2022-11-08 at 15 32 03" src="https://user-images.githubusercontent.com/846565/200607460-12a267b5-1275-4cd1-9866-f15985aa5316.png"> | <img width="1146" alt="Screenshot 2022-11-08 at 15 32 19" src="https://user-images.githubusercontent.com/846565/200607467-2dbc2736-2033-47d2-b582-4cbc6787eaa0.png"> |

Clearly it must still be possible to edit the post title without engaging the template editor first. For a solution we can look to the Site Editor, where the document name appears in the Top Bar, and custom templates can be renamed:

<img width="307" alt="Screenshot 2022-11-08 at 14 30 42" src="https://user-images.githubusercontent.com/846565/200608410-74c9e939-ab9d-4805-987d-775808bac036.png">

We can probably migrate this pattern across to the post editor as well. The result would be something like this:

<img width="1055" alt="Screenshot 2022-11-08 at 15 38 43" src="https://user-images.githubusercontent.com/846565/200609052-ccf3c7f6-c2b3-4268-8f2f-e7e1e48fc128.png">

When the document has no title, the label can probably read something like "Untitled Document".

<details><summary>Original issue</summary>I think it makes sense to move the page/post title into the top bar like we do for template names. Often we will want to add the Post title inside another block (e.g. Cover block), or in a column. 

This would give users more flexibility about where the Post/Page title appears:

![FSE-UX-flow](https://user-images.githubusercontent.com/5323259/94144423-65267480-fe71-11ea-92e8-f30d5768ee0f.gif)

Originally suggested by @paaljoachim here: https://github.com/WordPress/gutenberg/issues/20877#issuecomment-698309796</details>

**Comment by jameskoster** [comment]
It would also provide a way to change the post title when the post template does not include a post title block. Some post formats do this.

**Comment by scruffian** [comment]
Yeah, that's a common pattern in themes in general.

**Comment by paaljoachim** [comment]
It would be great to create movement in this issue.
The Site Editor uses the new method. Let's get a PR going for the Post Editor.

When the top toolbar is active it can be moved to a second row. The first row can contain the page/post title.
See this comment: https://github.com/WordPress/gutenberg/issues/20877#issuecomment-703064983

Let me know how I can help move this issue onward.
Thanks!

**Comment by jameskoster** [comment]
I agree it would be good to display the document title, and include the popover for editing the title & permalink.

But the interaction between the "Site" and "Page" buttons in the gif above isn't clear enough imo. I suggest we leave that out for now, since there will be (are already) other UI patterns for moving between template / content editing.

**Comment by paaljoachim** [comment]
Here is an update of the Figma file. 

https://user-images.githubusercontent.com/5323259/113604129-230e9480-9645-11eb-8fc6-4cf8a410a667.mp4

Btw when the top toolbar is active it is then automatically moved to the second line. As can also be seen when the browser has a smaller width. A mockup.

![Toolbar-2nd-row](https://user-images.githubusercontent.com/5323259/113604743-f7d87500-9645-11eb-8da7-4a87e11b03a3.jpg)

**Comment by jameskoster** [comment]
It might also be good to include the template information here, and a link to view all posts.


https://user-images.githubusercontent.com/846565/114733196-0d7f2600-9d3b-11eb-9a19-b6e7fa187c95.mp4

**Comment by paaljoachim** [comment]
I think we should break this down into iterations.

First version/iteration: (With a focus of atleast getting the fist iteration into WP 5.8.)
Let's begin by getting the page title into the top bar along with a drop down where one can rename the page title.

![Screen Shot 2021-04-22 at 13 46 46](https://user-images.githubusercontent.com/5323259/115709205-3756cf00-a371-11eb-8062-009e1c4becb8.png)


We need a developer for the above part. As it is a totally new concept for the Post Editor to have the page/post title in the top bar.

**Comment by vdwijngaert** [comment]
I'll try and have a go at this. Progress so far looks promising: 
![image](https://user-images.githubusercontent.com/772137/116256581-9c486580-a773-11eb-96db-ceb34bf120e7.png)

Here's at least two things that need some working out:

* How would we go about and replace the current `new post/page` workflow that focuses the empty post title block? 
* How should this work on a mobile viewport?

**Comment by jameskoster** [comment]
> How would we go about and replace the current new post/page workflow that focuses the empty post title block?

It may be fine to leave the post title on the canvas _for now_. Otherwise I imagine the text would read "Untitled Page" (or similar) until you click it. Clicking should shift focus directly to the title input.

**Comment by paaljoachim** [comment]
A couple of options.

![Add-a-title](https://user-images.githubusercontent.com/5323259/116260627-234b0d00-a777-11eb-97fe-efc762b9a0ab.jpg)


As In automatically add a placeholder showing the Post/Page title block. User adds it there and also sees the name added to the top bar. EDIT: This way a user are also able to remove the Post title block from the layout/canvas area.

----

Creating a new post/page the cursor blinks in the title field in the top bar where the placeholder says "Add page title".

![Page-title-field-top-bar](https://user-images.githubusercontent.com/5323259/116260350-e54de900-a776-11eb-8f58-c08e65352810.jpg)

@jasmussen Joen could you also take a look? Thank you.

**Comment by bobbingwide** [comment]
There could be a problem with this. In the Site Editor, without the post title,  it becomes difficult to edit the first block / blocks.
![image](https://user-images.githubusercontent.com/2474435/116427766-3a9ffe00-a83c-11eb-8992-a394e084bc35.png)

Workaround is to switch to having the toolbar at the top.

**Comment by jasmussen** [comment]
There's also this issue for the block toolbar when it's close to the top of the screen: https://github.com/WordPress/gutenberg/issues/29464

**Comment by jasmussen** [comment]
Having an interface in the top toolbar is something we've discussed for a while. The primary benefit is that it is the top of the hierarchy, so it clearly implies a document-level effect. The downside is that it can potentially take up a lot of valuable space. 

Going back to the intent of this ticket: _moving_ the editing of the page/post title to the top toolbar to result in a more WYSIWYG experience, we should be careful with, as the title in the canvas is a well-learned interface, any change will cause headaches for people who just blog.

But there is a flow to figure out. In the site editor, we need the top center dropdown interface to show document level metadata and attributes. It's present even in the new post-editor to template-editor switch:

![editing templates](https://user-images.githubusercontent.com/1204802/116522916-fa8d5980-a8d5-11eb-9c86-33d71595d459.gif)

Both of those are more accurate WYSIWYG than the post editor is at the moment. So it might be healthy to split the conversation into two flows:

- Is there a benefit to _duplicating_ the post/page title in a dropdown in the top toolbar, just like we have in the site editor? For example it feels like an obvious place for metadata such as permalink, maybe others.
- To address the original intent of this ticket, is there a good flow for showing a more accurate frontend rendition of the post you're editing, in-situ? I.e. show header and footer, hide post title unless also present in the template. Is there value in this, and where should it live? In a "View" dropdown?

Those two, when both addressed one by one, could help give us insights into whether it makes sense at all to remove the post title from the post editor when it's also missing from the post template. 

> How would we go about and replace the current new post/page workflow that focuses the empty post title block?
> How should this work on a mobile viewport?

If we focus not on removing the page/post title from the canvas, but instead creating the new dropdown interface to _also_ edit it there, it will help surface better answers to those questions. At the moment we can only make assumptions.

To that extent, I think Jay best outlines the longer term thinking around such a top toolbar interface in his post about [document attributes](https://jameskoster.design/2021/04/27/migrating-document-attributes-from-the-inspector-to-the-top-bar/): it's about better surfacing key document metadata in an obvious place. More accurate WYSIWYG preview of your post could benefit from that, but needs its own flow explored.

**Comment by jameskoster** [comment]
Good thoughts, I agree that splitting this in two makes sense. 

Also worth noting that while seeing the title on the canvas _does_ degrade the overall WYSIWYG experience, that degradation feels less impactful thanks to the edit template flow. We're now set up to provide a WYSIWYG of the post content in isolation, and a WYSIWYG of the post content in the context of its template where you're able to see the post title doing its thing.

A possible interim could be to leave the title on the canvas, but reduce its prominence?

| Before | After |
| --- | --- |
| <img width="1340" alt="Screenshot 2021-04-29 at 10 16 13" src="https://user-images.githubusercontent.com/846565/116529389-cf563a80-a8d4-11eb-8ae4-76609bfd8d59.png"> | <img width="1339" alt="Screenshot 2021-04-29 at 10 21 43" src="https://user-images.githubusercontent.com/846565/116529410-d54c1b80-a8d4-11eb-83eb-e3d0e072f5c5.png"> |

**Comment by vdwijngaert** [comment]
> A possible interim could be to leave the title on the canvas, but reduce its prominence?

That might work!

**Comment by jasmussen** [comment]
Even without any changes to the post title in the canvas — keep in mind, a lot of themes style it to match the template — I could see value in showing important meta attributes in the to dropdown.

**Comment by paaljoachim** [comment]
Additional thoughts.
Let's say we leave it mostly as it is today. With one big difference one changes the treatment of the title to use the Post Title block.

Step by step.
1. One sees the post/page as one is used to. Inserts the title.
2. Notices that there is a block toolbar associated with the title, and is now able to move the title, style or treat the title like any other block. Even remove it.
3. User removes the block, and the post/page title is automatically added to the top bar area.
4. User adds back in the Post title block.


https://user-images.githubusercontent.com/5323259/116930047-8816d380-ac5f-11eb-8e7f-a8b556413da6.mp4


https://www.figma.com/file/NE9Y71AMMxVkuLEFzCtoNa/Moving-Post-title-to-top-bar?node-id=0%3A1


What do you think @jameskoster , @jasmussen and @vdwijngaert ?

**Comment by vdwijngaert** [comment]
Hey @paaljoachim, appreciate the work you're putting into this, turns out it's quite complex :) 

> Notices that there is a block toolbar associated with the title, and is now able to move the title, style or treat the title like any other block. Even remove it.

That could work. However, when a template is being used, the location and styling of the title should probably be left to the template editor. Alternatively, we could have a slimmed down post title block that can open the current template (if any) in the template editor? This could be a sidebar panel, toolbar action, etc. 

But then, as @jameskoster stated, some templates [don't include a post title block](https://github.com/WordPress/gutenberg/issues/27093#issuecomment-768296305). 

So yeah, maybe we should leave the title unharmed for now? We could make it more obvious that it might not reflect how it would look on the front end. For now, we can just focus on making the title also editable in the toolbar.

**Comment by jameskoster** [comment]
Yeah in this case the post title is always going to be tied to a template, whether that is an "old" php template, or a block template. So allowing folks to manipulate (or even remove) it while editing a post will be misleading.

> For now, we can just focus on making the title also editable in the toolbar.

+1. We can address the on-canvas post title in a follow-up.

**Comment by amir2mi** [comment]
I think space between the left and right parts of the header gives identity to Gutenberg.

I have made this, what do you think?
![idea-demo](https://user-images.githubusercontent.com/64708228/117015028-a08df900-ad06-11eb-8f22-462e0cf43efa.gif)
Video: https://user-images.githubusercontent.com/64708228/116993399-abd42b00-acec-11eb-9cba-495f37256020.mp4

**Comment by paaljoachim** [comment]
That looks really cool, @amir2mi Amir! 
An interesting user interface. I am not sure how it fits into a more general title in the top bar UI though.

**Comment by amir2mi** [comment]
> I am not sure how it fits into a more general title in the top bar UI though.

Me too, just wanted to share the idea [those buttons can be hidden as well].
Can something like this be there when there is no title block? (like what you said conditional).
Is it a pain to distinguish which post I'm editing now on switching between tabs?

**Comment by jameskoster** [comment]
Earlier today I shared a concept that explores moving _template_ attributes to the top bar: https://github.com/WordPress/gutenberg/issues/31076#issuecomment-833590145

I decided to also explore how this pattern might be applied to _posts and pages_:

https://user-images.githubusercontent.com/846565/117320994-a90c3e00-ae84-11eb-842c-88906d16475d.mp4

I'm quite excited about this concept not only because it streamlines the inline editing experience, but also because it would enable us to greatly enhance the existing quick edit interface that exists in list view:

| Current | Future? |
| --- | --- |
| <img width="1500" alt="Screenshot 2021-05-06 at 16 04 58" src="https://user-images.githubusercontent.com/846565/117321298-ebce1600-ae84-11eb-920d-fb2bbf73898c.png"> | ![quick edit](https://user-images.githubusercontent.com/846565/117321908-71ea5c80-ae85-11eb-9184-86d8b8eb4946.png) |

**Comment by jameskoster** [comment]
Here's how we could potentially take the concept from my previous comment and strip it back to an MVP in the context of this issue:


https://user-images.githubusercontent.com/846565/117433516-736a6200-af23-11eb-82a5-705aa0c0a610.mp4



[Figma prototype](https://www.figma.com/proto/AFcjNvD8LRiNcL0fpH1CnS/Conceptual-Document-attributes?page-id=860%3A0&node-id=860%3A1&viewport=308%2C661%2C0.44032564759254456&scaling=scale-down).

**Comment by jasmussen** [comment]
I'm curious why it should be a modal as opposed to a dropdown?

**Comment by vdwijngaert** [comment]
> I'm curious why it should be a modal as opposed to a dropdown?

I was under the impression we were going to do a dropdown as well.

**Comment by jameskoster** [comment]
Modals scale better:

<img width="1647" alt="Screenshot 2021-05-07 at 13 56 37" src="https://user-images.githubusercontent.com/846565/117452764-1038f980-af3c-11eb-95a0-f62e872207b5.png">

We're using modals for names elsewhere:

<img width="380" alt="Screenshot 2021-05-07 at 13 57 44" src="https://user-images.githubusercontent.com/846565/117452909-38285d00-af3c-11eb-88d4-55cae492a4a6.png">

Modals are arguably more user-friendly on mobile. Popovers feel fragile since tapping outside accidentally will close the popover.

Finally, modals are portable. We can use this same interface to replace the ugly quick-edit feature in list view, as outlined in https://github.com/WordPress/gutenberg/issues/27093#issuecomment-833609029.

---

All that said, it doesn't have to be a modal just yet. It's only a wrapper after all. The UX is virtually identical both ways. My main argument for using a modal _now_ is that it paves the way for when we expand the interactions here to include other document attributes like template selection.

Edit: Here's the same flow, using a popover rather than a modal:


https://user-images.githubusercontent.com/846565/117454987-9ce4b700-af3e-11eb-9a63-c12de6ff2581.mp4

**Comment by jasmussen** [comment]
While I see the sense in putting document properties there, I'm not yet convinced we can put _all_ of it there. And the problem with the dialog is that it loses a bit of connection to the post title, whereas the dropdown has a clear connection to the title, giving a strong connection. Since both dialog and dropdown are technically modal (they trap tab inside and close with escape), the dropdown could become fullscreen on mobile still, even if it's a dropdown on desktop breakpoints.

It's possible that we will end up needing all the space of a dialog in the future, but I think for the near future we should start smaller, also literally 😅

**Comment by vdwijngaert** [comment]
Pushed changes including a working panel to edit the current entity title. Functionality works, but needs clean-up and polishing. :)

![image](https://user-images.githubusercontent.com/772137/117632814-bb84c100-b17d-11eb-9bda-7bb384b951b1.png)

**Comment by paaljoachim** [comment]
Here is the link to the PR which Koen @vdwijngaert is working on. It could use some feedback/advice.
https://github.com/WordPress/gutenberg/pull/31288

**Comment by paaljoachim** [comment]
As I logged into the Make WordPress Core web site to begin creating the Core Editor Summary. 
I noticed this:

https://user-images.githubusercontent.com/5323259/119130122-0f25c300-ba38-11eb-837a-ff3e568a5cd6.mp4

I have not seen the "Add title" text button before, and I am wondering where it comes from. Then I begin wondering if this is something new that has been added, and began thinking about how we can actually make use of it for moving the title to the top bar issue.

**Comment by paaljoachim** [comment]
A status update. (**Edited to clarify**)

**Part 1.**

The PR: [**WIP] Move title to header toolbar**
https://github.com/WordPress/gutenberg/pull/31288  Has stalled. I am hoping @vdwijngaert Koen is able to continue working on the PR. If someone is able to step in and take over that would be really helpful. Here one needs to duplicate how Full Site Editing adds the title to the top bar, and implement this for the post/page title.

![Add-title-to-top-bar](https://user-images.githubusercontent.com/5323259/137901255-31b6d3fb-9b91-4371-b8d0-0cab05244859.jpg)


--

**Part 2.**

The next PR needed to follow after the above PR. Is to remove the static post/page title we currently see in each post or page and replace it with the Post Title block.

![Add-title-to-top-bar](https://user-images.githubusercontent.com/5323259/137902696-a8e6d9a2-25f5-4cda-9932-d10bc018bcf2.jpg)

Both of the above features need a developer that can work on it.

As part of Full Site Editing it would be helpful to have both of the above features merged in time to be included into WordPress 5.9.

**Comment by Mamaduka** [comment]
The title in the top bar doesn't work well with long texts, as we discovered in #34512. This UI works in Site Editor since templates usually don't have long names.

I think @shaunandrews was [exploring](https://shaunandrews.com/2021/09/the-wordpress-editor-document-status-and-visibility/) moving a title in the Document sidebar.

**Comment by jameskoster** [comment]
I like that exploration a lot, but the sidebar is not always open, so I think we should at least _print_ the document title in the top bar as well. We can use [text-overflow](https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/) to truncate really long titles.

**Comment by Mamaduka** [comment]
I agree, just moving the title in the sidebar won't do it.

> We can use text-overflow to truncate really long titles.

We're using the `Text` component with `truncate` in Navigation Screen.

https://github.com/WordPress/gutenberg/blob/1e17b5a5b4a3117e843dde015c128361c5ae059e/packages/edit-navigation/src/components/header/menu-actions.js#L42-L51

**Comment by jameskoster** [comment]
Oh nice, I didn't know about that. We should be able to do the same thing here, right?

**Comment by Mamaduka** [comment]
Yes, we can use the same method here.

**Comment by shaunandrews** [comment]
Here's a example of how it could work with the title being editable in the sidebar, while also having the title in the top bar as a shortcut for opening the sidebar and focusing the title input:

https://user-images.githubusercontent.com/191598/138347448-f5efb4f1-95a6-4ecd-83d4-eaa3b11a6f9e.mp4

If we move to use Tools Panel for the document sidebar, we could also include an explicit option in the menu to rename the title:

<img width="997" alt="image" src="https://user-images.githubusercontent.com/191598/138348287-e6601094-46af-426e-b1e5-7e3536fc4c69.png">

**Comment by jameskoster** [comment]
Just noting that we probably shouldn't expose the Post Title toolbar like this, since all of the properties therein are tied to the template rather than the post itself. IE it's not possible to move, transform, change alignment, etc in this context.

In other words we'll need to come up with another way to toggle the visibility of the title. I think we can probably do that in a follow-up though, as I'm not sure we even need to display the title on the canvas at all once it is visible (and editable) via the Top Bar.

**Comment by vdwijngaert** [comment]
Okay guys, I'm back from my little Gutenberg-burnout, it seems :) 

I resumed work on #31288, and it looks like it is functionally ready for [part 1 as described](https://github.com/WordPress/gutenberg/issues/27093#issuecomment-946607869) by @paaljoachim.  I updated the code to reflect the changes made to `DocumentActions` from the Site Editor package. 

I also had to go ahead and update the layout of the header as well, to resemble the Site Editor header a bit more, otherwise the title would have always been a bit _off-center_. I still need to go trough the styling to remove some things that have been rendered obsolete because of this.

**Comment by apeatling** [comment]
> If we move to use Tools Panel for the document sidebar, we could also include an explicit option in the menu to rename the title

I like the idea of making the post title editable in the sidebar. With the ToolsPanel option I'd prefer to see it as a setting you can toggle like any other, to avoid there being two ways to do things.

**Comment by paaljoachim** [comment]
I assume we can go with a 1st version mentioned in my above comment: 
https://github.com/WordPress/gutenberg/issues/27093#issuecomment-946607869

Currently when I attended Brian's @bgardner and Nick's @ndiego Full Site Editing chat yesterday. 
Brian showed the cumbersome work around in creating a page template and in the page template he removed the Post title, and saved the template. So that all pages that use the template will not show the page title on the frontend. 

Hopefully someone will be able to either take over @vdwijngaert 's PR or create a new. I look forward to having a better consistency between the Site Editor and Post Editor.
Thanks!

**Comment by jameskoster** [comment]
The case for making this change is imo stronger than ever.

1. We don't want to display the document title on the canvas because it is not a true visual representation of the frontend.
2. Splitting document / block settings goes hand-in-hand with [splitting block appearance controls / settings](https://github.com/WordPress/gutenberg/issues/40204).
3. We've already implemented this pattern in the Site Editor for handling template settings:

<img width="347" alt="Screenshot 2022-05-10 at 12 39 51" src="https://user-images.githubusercontent.com/846565/167620227-4569c3de-d898-4ee0-9650-e0b4cb75334e.png">

Here's a quick example of how things could work if we combined the drilldown pattern from Global Styles with the excellent work folks have done to [refine the status and visibility panel](https://github.com/WordPress/gutenberg/issues/39082).

https://user-images.githubusercontent.com/846565/167620118-5149f646-c272-4a25-bf9b-154bdd6f41ff.mp4

**Comment by jameskoster** [comment]
I've updated the OP to include designs that we can put into practise now.

**Comment by jasmussen** [comment]
Nice, thank you Jay. I went ahead and _hid_ some outdated conversation just so devs can better focus on the contents of the original issue.

**Comment by artemiomorales** [comment]
I can take this on. I'm currently working to implement a popover to update just the title so we have something working to look at. Just let me know if I should be considering anything else!

> I've updated the OP to include designs that we can put into practise now.

Question for @jameskoster: To clarify, do you mean that you've updated the original post?

**Comment by jameskoster** [comment]
> do you mean that you've updated the original post?

That's correct :)

**Comment by artemiomorales** [comment]
Hello all, I created a pull request for this issue: https://github.com/WordPress/gutenberg/pull/46135

It's the most basic implementation for now. Would love to get additional eyes on it and see what you all think ✨

**Comment by alexstine** [comment]
Hello all, I just reviewed the PR and gave some technical/accessibility feedback. Keeping in mind that I can't see the screenshots, I am interested to see how close the PR got to the actual visual design. From an UX perspective, it greatly worries me that we continue to box up the editor. It is going to get harder and harder to provide context around that as we'll soon require a bunch of random/very generic button labels to convey where we've decided to hide certain functionality/options. I am not saying wide open always works better, but I am not sure the post title hidden in an accordion-like drop down was the right way to go here.

In my opinion, I think it would be nice to leave the title field where it currently is. It is a nice buffer between main document title and the start of the editing fun.

Thoughts?

Thanks.

**Comment by jameskoster** [comment]
The issue with the current placement of the title field is that it breaks the WYSIWYG. The proposed solution re-uses existing Site Editor features (updating the title of a custom template), but I assume those also suffer from the same accessibility issues?

An alternative option could be to simply move the title field into the top bar? Hopefully we could position it in the DOM so that it still acts as a buffer?


https://user-images.githubusercontent.com/846565/204784931-166aff69-952c-49b6-9c5a-314d7aac9a1e.mp4

**Comment by alexstine** [comment]
@jameskoster Might be best to just move the field in to the top bar. I am not sure hiding it in a drop down is really necessary at this point. All my other A11Y feedback on the PR still applies. Need to figure out how to not mess up the DOM structure.

Thanks.

**Comment by jameskoster** [comment]
Assuming we can get the other a11y details right, I think it would be okay to start with that. Thanks for the detailed feedback as always.

Edit: to clarify, which of these points would remain if the input was moved into the top bar?

> 1. "Edit document details" button should probably not be a sub menu in the event that it simply shows the title field.
> 2. The title field is rendered at the bottom of the DOM just below the block breadcrumbs. The title field needs to appear in relation to the button.
> 3. "Edit document details" does not say anything about editing the title. My problem is, we're going to start splitting hairs until no one knows how to find anything in the editor. From a general UX perspective, hiding everything in toggles can't be helping us long term.
> 4. Once you are done editing the title, there is no easy way back to the button that moved focus to the title field.
> 5. It is not clear to me how to save the title changes once done editing.

The first, third and fourth issues should be resolved because the "Edit document details" button wouldn't be required any more.

Issue #2 should be easy to address, we can position the title input anywhere in the top bar and use css for ordering / styling.

For the fifth issue, how do you know how to save the title changes currently?

**Comment by joedolson** [comment]
I think that Alex has a good handle on the accessibility of the title interface. I'm comfortable in principle with moving the title out of the post editor; it was a separate field in the classic editor, and keeping it separate seems totally reasonable to me. 

My primary concern is making sure that the title has sufficient prominence, so that it's easy to spot at a glance which document you're editing and that there's no truncation of the title, so that the entire post title is always visible. 

Having the title directly editable in the top bar makes sense to me. I can also imagine that this layout could speed things up in a work flow where you're stubbing out a large number of documents to be filled out later, since you wouldn't need to enter the post editor in order to publish a stub that only had a title.

**Comment by artemiomorales** [comment]
I agree with @joedolson. As a user, I wouldn't know the title was immediately editable without some kind of indication, so having it as an input makes sense to me. I updated the [pull request](https://github.com/WordPress/gutenberg/pull/46135) with that implementation so we can all see what that looks like as we keep the discussion going.

> 5. It is not clear to me how to save the title changes once done editing.

I believe having it as an input also addresses this fifth issue raised by @alexstine. If the title is an input, I feel that folks could intuit that one needs to save the post in order to persist edits to it.

**Comment by annezazu** [comment]
I wanted to note for tracking purposes that the PR to address this has been closed in favor of a potential future, broader fix in https://github.com/WordPress/gutenberg/issues/44461 or https://github.com/WordPress/gutenberg/issues/27847. Let's keep this open for now to ensure this does indeed happen as part of Phase 2 work.

**Comment by Mamaduka** [comment]
@annezazu, @ndiego, based on the discussion in PR https://github.com/WordPress/gutenberg/pull/46135#issuecomment-1362924924, I believe this path was deprioritized in favor of a broader solution.

I think we can remove the issue from WP 6.2 project board.

**Comment by ndiego** [comment]
Yeah, at the very least, it does not look like a solution will make it into 6.2 at this point. I'll remove it.

**Comment by Mamaduka** [comment]
Can we close this issue? The editor now has a Document bar displaying the title + new rename action in the sidebar.

**Comment by jasmussen** [comment]
I think there are opportunities for putting a rename action in the title area again, though the precise mechanism for doing so needs a little thought as it also opens the command palette. I don't know if that means we keep this issue open or not.

**Comment by Mamaduka** [comment]
I personally like the current design and functionality of the document bar. @jasmussen, as you mentioned, the click is already attached to the command palette. In the future, maybe renaming action can be a command.

@youknowriad, any thoughts?

**Comment by youknowriad** [comment]
Agreed that a contextual command seems like a natural approach here.

**Comment by krokodok** [comment]
I think the main point of this issue was not only to have the option to edit the post/page title somewhere outside the content but this:

> … it might be time to consider _not_ displaying the post title within the canvas …

**Comment by Mamaduka** [comment]
Closing this as resolved.

**Comment by paaljoachim** [comment]
A little messy brainstorming here.... Not sure where to put this right now.
For me this issue was/is about moving the title out of the canvas area . It is the only area that is not editable breaking consistency that we have in which all content on the canvas are blocks.

What if we make the title into a block? Making it possible for the user to remove the title and just keep the title in the right sidebar settings... Hmm not sure about this one. Is that good enough even though? 

Currently if one does not want the title to be seen one needs to go to a template and there remove the title from the template.

Keeping something in the canvas that is not a block just does not seem right. I have to over and over again explain this to customers.

I have said something like this...
The title is actually not a block and can not be removed and not easily styled. There might be specific pages where we want to show the title and other pages where it is not needed. For instance we do not want to show the Contact us title, but on the About us we want that to show. Well then we have to create a heading as a pretend title block for the pages that want a title. Remove the title from the pages template. So that for pages where we want the title to be seen then we have the Heading block with the same name as the title. Even if there are now two titles on the page. One that is not seen and the other which is seen when viewing the web site. Also blocks can easily be styled directly in the page.

I try and not create confusion when explaining this to customers but this one is fairly confusing. 


I want to say this.
The title seen can be removed if one does not want it seen for the specific page. The title is still listed in the top middle area (which now will likely create some conflict with the command palette) and in the right page settings area.

**Comment by paaljoachim** [comment]
Associated issue:
Add a new document button and menu to Site Editor to allow for easier switching between pages and templates
https://github.com/WordPress/gutenberg/issues/74045
